src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/Et3VettingService.java

```diff
@@ -0,0 +1,93 @@
+package uk.gov.hmcts.ethos.replacement.docmosis.service;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.webjars.NotFoundException;
+import uk.gov.hmcts.et.common.model.bulk.types.DynamicFixedListType;
+import uk.gov.hmcts.et.common.model.ccd.CaseData;
+import uk.gov.hmcts.et.common.model.ccd.items.RespondentSumTypeItem;
+import uk.gov.hmcts.et.common.model.ccd.types.Et3VettingType;
+
+import java.util.Optional;
+
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.YES;
+import static uk.gov.hmcts.ethos.replacement.docmosis.helpers.Helper.objectMapper;
+
+@Slf4j
+@Service()
+public class Et3VettingService {
+    /**
+     * Moves ET3 Vetting related fields off of CaseData and onto the relevant respondent.
+     * @param caseData The object containing case data
+     */
+    public void saveEt3VettingToRespondent(CaseData caseData) {
+        String respondentName = caseData.getEt3ChooseRespondent().getSelectedLabel();
+        RespondentSumTypeItem respondent = getRespondentForCase(respondentName, caseData);
+
+        respondent.getValue().setEt3Vetting(copyEt3FieldsFromCaseDataToRespondent(caseData));
+        updateValuesOnObject(caseData, new Et3VettingType());
+
+        respondent.getValue().setEt3VettingCompleted(YES);
+    }
+
+    /**
+     * Restores respondent specific ET3 Vetting related data to the caseData object for editing within ExUI.
+     * @param caseData The object containing case data
+     */
+    public void restoreEt3VettingFromRespondentOntoCaseData(CaseData caseData) {
+        String respondentName = caseData.getEt3ChooseRespondent().getSelectedLabel();
+        RespondentSumTypeItem respondent = getRespondentForCase(respondentName, caseData);
+
+        Et3VettingType et3Vetting = getEt3VettingTypeForRespondent(respondent);
+        DynamicFixedListType et3ChooseRespondent = caseData.getEt3ChooseRespondent();
+        updateValuesOnObject(caseData, et3Vetting);
+        caseData.setEt3ChooseRespondent(et3ChooseRespondent);
+    }
+
+    /**
+     * Updates et3 related fields on caseData with values from an Et3VettingType.
+     * @param caseData to update
+     * @param et3VettingType to bring in values from
+     */
+    public void updateValuesOnObject(CaseData caseData, Et3VettingType et3VettingType) {
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+        try {
+            mapper.updateValue(caseData, et3VettingType);
+        } catch (JsonMappingException e) {
+            log.error(String.format("Failed to restore et3 data for case %s because %s",
+                caseData.getEthosCaseReference(), e.getMessage()));
```

Comment:
Just e, not only the message

I.e.
log.error(String.format("Failed to restore et3 data for case %s",
                caseData.getEthosCaseReference(), e))